qiskit/circuit/bit.py

```diff
@@ -49,33 +54,50 @@ def _update_hash(self):
     @property
     def register(self):
         """Get bit's register."""
+        if (self._register, self._index) == (None, None):
+            raise CircuitError('Attmped to query register of a new-style Bit.')
```

Comment:
I would probably say registerless bit instead of new-style. I think this context around new-style is temporary.

Reply:
Agree the context is temporary and only for the transition. I went with old- and new-style over registerless because the new `Bit`s can still be a part of a register, which might be confusing. There are only a few places that call out the differences, so I'm happy to update if you think it's clearer.

Reply:
It would be clearer to me. In my head this kind of bit is always registerless even if it's part of a register it's still a registerless bit right (the bit doesn't have a register in its context) but instead the register contains the bits? It's splitting hairs but that's how I would think about it. But, I'll defer to others because what's clearer to me isn't necessarily clear for everyone else and this isn't a big deal either way.

Reply:
Okay, good point. For now, I'll leave these in place and when we're writing documentation/release notes, we can decide how best to phrase it (and update the warnings accordingly).